### ISSUE TITLE
Preventing overflow fails when using arrow margin

### ISSUE THREAD

**Issue by renspoesse** [popperjs/popper.js#806]
### CodePen demo

https://jsfiddle.net/xp7syfw5/9/

### Steps to reproduce the problem

1.  Create a tooltip with an arrow element (similar to the examples on popper.js.org), and give the tooltip rounded corners. It has been suggested (https://github.com/FezVrasta/popper.js/issues/343) to apply a margin to the arrow to prevent it from reaching the rounded corners.
2.  Create a fairly small reference element.
3.  Use `preventOverflow` to keep the popover within a `boundariesElement`, in this case without any padding. The distance between the `boundariesElement` and the reference should be just large enough to position the popover within the container after accounting for the arrow margin.

### What is the expected behavior?

The popover should be positioned just against the left border of the `boundariesElement`. The arrow should end up in the center of the reference element, because there's enough space for half the arrow + its left margin between the `boundariesElement` left border and the reference center.

![image](https://user-images.githubusercontent.com/15845884/61489305-019cdf00-a970-11e9-8130-56211f8fb985.png)

### What went wrong?

The popover overflows the container on the left.

![image](https://user-images.githubusercontent.com/15845884/61489192-b5ea3580-a96f-11e9-8e6c-47bc54ed2be9.png)

### Any other comments?

Removing the margin on `.arrow-container` seems to fix the problem.

**Comment by Mesoptier** [comment]
I have run into the same issue. The main problem is that `flip` only checks whether the left side of the popper overflows if the placement is also `left` (and similar for the other directions).

Locally, I have removed the `placement === ?` checks, making `flip` also flip when the popper is out of bounds on the other edges.

https://github.com/FezVrasta/popper.js/blob/9e94c9814250e6ca6db0bd7d6f3bd487b963491b/packages/popper/src/modifiers/flip.js#L90-L94

---

N.B. Another issue arised after applying this fix. Though this might only be a problem for me. The popper would start flipping (heh) out and change its side on each update. I fixed this by forcibly setting the `x-placement` attribute before the other modifiers are run with this custom `reset` modifier:

```js
reset: {
    order: 1,
    enabled: true,
    fn(data) {
        // This makes sure the correct CSS is applied before getting the size of the popper
        data.instance.popper.setAttribute(
            'x-placement',
            data.placement
        );
        return data;
    },
},
```
